src/threading.c

```diff
@@ -774,6 +774,7 @@ void _jl_mutex_wait(jl_task_t *self, jl_mutex_t *lock, int safepoint)
         lock->count++;
         return;
     }
+    JL_TIMING(LOCK_SPIN, LOCK_SPIN);
```

Comment:
Is this that helpful? I would rather use the lock mechanism in ITTAPI.

Reply:
When CPU time is being viewed, it's nearly impossible to see what's part of the wait and what isn't since the CPU time covers up the waiting period. Also, it helps distinguish between waiting for a task and spinning on a lock .

![image](https://user-images.githubusercontent.com/34727397/233465330-33b3ae4a-f60c-4f81-9b5a-0ce663cd26e3.png)
![image](https://user-images.githubusercontent.com/34727397/233465439-26ed9d1e-048d-4dc1-a094-f807634d5f71.png)
![image](https://user-images.githubusercontent.com/34727397/233465627-017b15ba-b531-4d1b-b9d4-c488403f1d25.png)

The dark green regions represent running, and the light green regions represent waits.

Reply:
Let's take a couple minutes to discuss this on the #profiling call today, if we can

I'm having trouble understanding this chart, since I'm not a VTune user

Reply:
Profiling call decided this was a useful feature, so it'll stay.